server/rest_slicer_cli.py

```diff
@@ -826,14 +829,163 @@ def genRESTEndPointsForSlicerCLIsInDocker(info, restResource, dockerImages):
     info
     restResource : str or girder.api.rest.Resource
         REST resource to which the end-points should be attached
-    dockerImages : str or list of str
-        A single docker image or a list of docker images
+    dockerImages : a list of docker image names
+
 
     Returns
     -------
 
     """
+    dockerImages
+    # validate restResource argument
+    if not isinstance(restResource, (str, Resource)):
+        raise Exception('restResource must either be a string or '
+                        'an object of girder.api.rest.Resource')
+
+    # validate dockerImages arguments
+    if not isinstance(dockerImages, (str, list)):
+        raise Exception('dockerImages must either be a single docker image '
+                        'string or a list of docker image strings')
+
+    if isinstance(dockerImages, list):
+        for img in dockerImages:
+            if not isinstance(img, str):
+                raise Exception('dockerImages must either be a single '
+                                'docker image string or a list of docker '
+                                'image strings')
+    else:
+        dockerImages = [dockerImages]
+
+    # create REST resource if given a name
+    if isinstance(restResource, str):
+        restResource = type(restResource,
+                            (Resource, ),
+                            {'resourceName': restResource})()
+
+    restResourceName = type(restResource).__name__
+
+    # Add REST routes for slicer CLIs in each docker image
+    cliList = []
+
+    for dimg in dockerImages:
+        # check if the docker image exists
+
+        id = localDockerImageExists(dimg)
+        if id is None:
+            id = pullDockerImage(dimg)
+        if id is None:
+            raise ValueError("The image %s does not exist locally or "
+                             "on the default registry" % dimg)
+        # get CLI list
+        cliListSpec = localDockerImageCLIList(dimg)
+        if cliListSpec is None:
+            raise ValueError("Could not get the cli list for the img %s most "
+                             "likely the docker image entrypoint does "
+                             "not accept the argument --list_cli" % dimg)
+        # pprint.pprint(cliListSpec)
+
+        cliListSpec = json.loads(cliListSpec)
 
+        # Add REST end-point for each CLI
+        for cliRelPath in cliListSpec.keys():
+            cliXML = localDockerImageclixml(dimg, cliRelPath)
+            if cliXML is None:
+                raise ValueError("Could not get the cli for the img %s cli %s"
+                                 " most likely the docker image entrypoint does"
+                                 " not accept the argument %s --xml" %
+                                 (dimg, cliRelPath, cliRelPath))
```

Comment:
Again you can move lines 892-896 (that check if `cliXML is None` and raise an exception) into the function `getDockerImageCLIXML`. That function should report an error if one occurs inside it. Also it would reduce cyclomatic complexity without sacrificing any readability.